sdk/identity/Azure.Identity/src/CredentialPipeline.cs

```diff
@@ -2,29 +2,38 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
 using Azure.Core;
+using Azure.Core.Diagnostics;
 using Azure.Core.Pipeline;
 using Microsoft.Identity.Client;
 
 namespace Azure.Identity
 {
    internal class CredentialPipeline
     {
-        private static readonly Lazy<CredentialPipeline> s_Singleton = new Lazy<CredentialPipeline>(() => new CredentialPipeline(new TokenCredentialOptions()));
+        private static readonly Lazy<CredentialPipeline> s_singleton = new Lazy<CredentialPipeline>(() => new CredentialPipeline(new TokenCredentialOptions()));
+
+        private static readonly AsyncLocal<IScopeHandler> s_scopeHandler = new AsyncLocal<IScopeHandler>();
```

Comment:
Can we avoid the AsyncLocal?

Reply:
All credentials share the same pipeline which is owned by factory. Let me try to hold the handler in `CredentialPipeline` itself.

Reply:
It works, but there is a problem with concurrent calls. I can address it in a fix for https://github.com/Azure/azure-sdk-for-net/issues/13044